subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -319,63 +284,46 @@ DefaultJavaCompileSpec createSpec() {
     }
 
     private boolean isToolchainCompatibleWithJava8() {
-        return getCompilerTool().get().getMetadata().getLanguageVersion().canCompileOrRun(8);
+        return getJavaCompiler().get().getMetadata().getLanguageVersion().canCompileOrRun(8);
     }
 
     @Input
     JavaVersion getJavaVersion() {
-        return JavaVersion.toVersion(getCompilerTool().get().getMetadata().getLanguageVersion().asInt());
+        return JavaVersion.toVersion(getJavaCompiler().get().getMetadata().getLanguageVersion().asInt());
     }
 
     private DefaultJavaCompileSpec createBaseSpec() {
-        final ForkOptions forkOptions = compileOptions.getForkOptions();
-        if (javaCompiler.isPresent()) {
-            applyToolchain(forkOptions);
-        }
-        return new DefaultJavaCompileSpecFactory(compileOptions, getToolchain()).create();
-    }
-
-    private void applyToolchain(ForkOptions forkOptions) {
-        final JavaInstallationMetadata metadata = getToolchain();
+        ForkOptions forkOptions = compileOptions.getForkOptions();
+        JavaInstallationMetadata metadata = getJavaCompiler().get().getMetadata();
         forkOptions.setJavaHome(metadata.getInstallationPath().getAsFile());
-    }
-
-    @Nullable
-    private JavaInstallationMetadata getToolchain() {
-        return javaCompiler.map(JavaCompiler::getMetadata).getOrNull();
+        forkOptions.setExecutable(null);
```

Comment:
I vaguely remember there was a behavior that forced the CommandLineCompiler. I think we are losing this by this change, right?